src/app/shopping-cart/shopping-cart.component.ts

```diff
@@ -14,8 +14,8 @@ export class ShoppingCartComponent implements OnInit {
   constructor(private shoppingCartService: ShoppingCartService) { }
 
   ngOnInit(): void {
-    this.shoppingCart = JSON.parse(localStorage.getItem("shoppingCart"));
-    this.total = JSON.parse(localStorage.getItem("total"));
+    let storage = this.shoppingCartService.getLocalStorage();
```

Comment:
Same as I said before - you should not expose implementation details of the data layer, outside the data layer, as is normal MVC.

The component does not care that you're using local storage, API, etc. You should only be interfacing with instances / models in the component.